src/main/java/Duke.java

```diff
@@ -1,10 +1,201 @@
+        return newTask;
+    }
+    public static void mainLogic(Scanner sc, ArrayList<Task> lst) {
+
+        while (true) {
+            String input = sc.nextLine();
+            String firstWord = input.split(" ")[0];
+            String output = "";
+            if (input.equals("bye")) {
+                saveToDisk(lst);
+                output = "Bye. Take care!";
+            } else if (input.equals("list")) {
+                output += "Here are the tasks in your list:";
+                for (int i = 0; i < lst.size(); i ++) {
+                    Task currentTask = lst.get(i);
+                    String num = Integer.toString(i + 1);
+                    output += "\n" + num + "." + currentTask;
+                }
+            } else if (firstWord.matches("done|delete")) {
+                String[] splitted = input.split("\\s+");
+                int taskIndex = Integer.parseInt(splitted[1]) - 1;
+                Task selectedTask = lst.get(taskIndex);
+                if (firstWord.equals("done")) {
+                    selectedTask.markAsDone();
+                    output += "Nice! I've marked this task as done:\n  " + selectedTask;
+                } else {
+                    lst.remove(taskIndex);
+                    output += "Noted. I've removed this task:\n  " + selectedTask;
+                }
+                output += "\nNow you have " + lst.size() + " tasks in the list.";
+            } else if (firstWord.matches("todo|deadline|event")) {
+                Task newTask;
+                try {
+                    newTask = createTask(firstWord, input);
+                    lst.add(newTask);
+                    output += "Got it. I've added this task:\n  " + newTask;
+                    output += "\nNow you have " + lst.size() + " tasks in the list.";
+                    printMessage(output);
+                } catch (DukeException e) {
+                    output += "☹ OOPS!!! " + e.getMessage();
+                    printMessage(output);
+                } finally {
+                    mainLogic(sc, lst);
+                }
+            } else {
+                try {
+                    invalidInput();
+                } catch (DukeException e) {
+                    output += "☹ OOPS!!! " + e.getMessage();
+                    printMessage(output);
+                } finally {
+                    mainLogic(sc, lst);
+                }
+            }
+            printMessage(output);
+        }
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String greetings = "Hello! I'm Duke, your personal assistant.\nWhat can I do for you?";
+        printMessage(greetings);
+
+        ArrayList<Task> lst = new ArrayList<>();
```

Comment:
A better practice might be to split the instantiation of "lst".

After your import statements, it could be List<Task> lst;
Then in your Duke constructor: this.lst = new ArrayList<>();